### ISSUE TITLE
Access-Control-Allow-Origin erros using the send method

### ISSUE THREAD

**Issue by PedroPMS** [mikehaertl/php-tmpfile#26]
Hello,

I have been using this library as part of a project in conjunction with phpwkhtmltopdf library.

Whenever I use the send method after creating the pdf, a cors error occurs because there is no Access-Control-Allow-Origin header, an alternative would be interesting to be able to include some headers before the response to the client. The same to phpwkhtmltopdf.

**Comment by mikehaertl** [comment]
Would an extra parameter `$headers` help? The signature of the send method would then be:

```
send($filename = null, $contentType = null, $inline = false, $headers = array())
```
This way you could also override some of the default headers like `Cache-Control`, etc.

**Comment by PedroPMS** [comment]
Yes, this will be fantastic!

**Comment by mikehaertl** [comment]
@PedroPMS I've prepared a MR for this change. Could you maybe help test it and check if it works for you? You'd have to check this standalone outside of phpwkthmltopdf, though.

The new features:
* Pass headers as 4th arg to `send()` in the form `['Header-name' => 'header value', ...]`
* Define default headers in `File::$defaultHeaders` 
  But be careful here: don't override default values, instead add them like `File::$defaultHeaders['MyHeader'] => 'Some value';`
* Override auto-generated headers like  `['Content-Length' => 12345, ...]`
* Suppress auto-generated headers with `['Content-Length' => false, ...]`

**Comment by PedroPMS** [comment]
I did some tests and it worked very well. Either using defaultHeaders or passing to the send method.

![image](https://user-images.githubusercontent.com/48099126/109536054-03aba700-7a9c-11eb-884b-a998c35777d0.png)
![image](https://user-images.githubusercontent.com/48099126/109536072-0908f180-7a9c-11eb-9823-98bed4fccbdb.png)


Thank you very much!

**Comment by PedroPMS** [comment]
Override and suppress auto-generated also worked very well.

**Comment by mikehaertl** [comment]
@PedroPMS I've released 1.2.1 containing this change. I've also opened a MR over at phpwkhtmltopdf: https://github.com/mikehaertl/phpwkhtmltopdf/pull/367

As it's hard to test these things with phpunit, maybe you could help again? It would be great if you could comment on the issue there: https://github.com/mikehaertl/phpwkhtmltopdf/issues/366
